lib/slide_hub/cloud/engine/aws.rb

```diff
@@ -0,0 +1,280 @@
+                  "https://#{@config.image_bucket_name}.s3-#{@config.region}.amazonaws.com"
+                end
+          url
+        end
+
+        def self.upload_endpoint
+          url = if @config.region == 'us-east-1'
+                  "https://#{@config.bucket_name}.s3.amazonaws.com"
+                else
+                  "https://#{@config.bucket_name}.s3-#{@config.region}.amazonaws.com"
+                end
+          url
+        end
+
+        def self.s3_host_name
+          s3_host_name = if @config.region == 'us-east-1'
+                           's3.amazonaws.com'
+                         else
+                           "s3-#{@config.region}.amazonaws.com"
+                         end
+          s3_host_name
+        end
+
+        ## SQS
+        def self.sqs
+          @sqs ||= Aws::SQS::Client.new(region: @config.region)
+        end
+
+        def self.send_message(message)
+          resp = self.sqs.send_message({
+            queue_url: @config.sqs_url,
+            message_body: message,
+          },)
+          resp
+        end
+
+        def self.receive_message(max_number = 10)
+          response = self.sqs.receive_message(queue_url: @config.sqs_url, visibility_timeout: 600, max_number_of_messages: max_number)
+          # see http://docs.aws.amazon.com/sdkforruby/api/Aws/SQS/Client.html#receive_message-instance_method
+          result = SlideHub::Cloud::Queue::Response.new
+          unless (!response || !response.respond_to?(:messages) || response.messages.count == 0)
+            response.messages.each do |msg|
+              result.add_message(msg.message_id, msg.body, msg.receipt_handle)
+            end
+          end
+          result
+        end
+
+        def self.delete_message(message_object)
+          resp = self.sqs.delete_message({
+            queue_url: @config.sqs_url,
+            receipt_handle: message_object.handle,
+          },)
+          resp
+        end
+
+        def self.batch_delete(entries)
+          # self.sqs.delete_message_batch(queue_url: @config.sqs_url, entries: entries)
+          result = true
+          entries.each do |entry|
+            result = self.delete_message(entry)
+          end
+          result
+        end
+
+        ## S3
+        #
+        def self.upload_files(bucket, files, prefix)
+          files.each do |f|
+            Aws::S3::Client.new(region: @config.region).put_object(
+              bucket: bucket,
+              key: "#{prefix}/#{File.basename(f)}",
+              body: File.read(f),
+                acl: 'public-read',
+                storage_class: 'REDUCED_REDUNDANCY',
+            ) if File.exist?(f)
+          end
+        end
+
+        def self.get_file_list(bucket, prefix)
+          resp = Aws::S3::Client.new(region: @config.region).list_objects({
+            bucket: bucket,
+            max_keys: 1000,
+            prefix: prefix,
+          },)
+          files = []
+          resp.contents.each do |f|
```

Comment:
__[reek]__
* [UncommunicativeVariableName] has the variable name 'f' [:ref](https://github.com/troessner/reek/blob/master/docs/Uncommunicative-Variable-Name.md)